lib/OpenQA/Worker/WebUIConnection.pm

```diff
@@ -0,0 +1,433 @@
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::Worker::WebUIConnection;
+use Mojo::Base 'Mojo::EventEmitter';
+
+use OpenQA::Utils qw(log_error log_debug log_warning log_info feature_scaling rand_range logistic_map_steps);
+use OpenQA::Constants qw(WEBSOCKET_API_VERSION MAX_TIMER MIN_TIMER);
+use OpenQA::Worker::CommandHandler;
+
+use Mojo::IOLoop;
+
+has 'webui_host';         # hostname:port of the web UI to connect to
+has 'url';                # URL of the web UI to connect to - initially deduced from webui_host (Mojo::URL instance)
+has 'ua';                 # the OpenQA::Client used to do connections
+has 'status';             # the status of the connection: new, registering, establishing_ws, connected, failed, disabled
+has 'worker';             # the worker this client belongs to
+has 'worker_id';          # the ID the web UI uses to track this worker (populated on registration)
+has 'testpool_server';    # testpool server for this web UI host
+has 'working_directory';  # share directory for this web UI host
+has 'cache_directory';    # cache directory for this web UI host
+has 'websocket_connection'
+  ; # the websocket connection to receive commands from the web UI and send the status (Mojo::Transaction::WebSockets instance)
+#has 'command_handler'; # handles commands from the web UI received via the websocket connection
+has 'webui_host_population';
+has 'send_status_interval'
+  ;    # interval for overall worker status updates; automatically set when needed (unless set manually)
```

Comment:
broken `;` better move comments on top of the corresponding variables then. Might need separating blank lines then though